### DISCUSSION TITLE
Allow to specify folders for view files in "add" commands

### DISCUSSION THREAD

**Discussion by exophunk** [studio1902/statamic-peak-commands#13] [Ideas]
I think the Peak CLI commands are awesome and really helpful to speed things up. While I'm quite happy with the basic Peak set up, I do like to add some more structure to my view files by nesting them into folders:

```
views/
    components/
        bard_sets/
        page_builder/
        layout/
        some_thing/
    collections/
    emails/
    ...
```

This works well and I only need to tweak the Peak setup a little, e.g. adjust the page builder block partial tag: 
```antlers
{{ partial src="components/page_builder/{type}" }}`
```
But changing the view folder structure breaks the scaffolding commands like `php artisan statamic:peak:add-block` because the folders are [hardcoded](https://github.com/studio1902/statamic-peak-commands/blob/main/src/Commands/AddBlock.php#L68).

I think it would be handy if there is a way to customize the folders. I see two ways how to do this and I'm not sure yet which one is better:

1. Ask for folder in Command
Simply add another question to the `add` commands asking for the folder with a default:
```php
$this->viewpath = $this->ask('What should be the path for this block view file?', 'page_builder');
// ...
File::put(base_path("resources/views/{$this->viewpath}/_{$this->filename}.antlers.html"), $contents);
```
✅ Easy to implement, Atomic, no side effects
✅ Adjustable for each block added
❌ More questions/steps
❌ Need to answer every time

2. Create a config file with view paths
Add a publishable config file `peak-cli.php` where all the default view paths are defined and one can customize it there:

```php
return [
    'view_folders' => [
        'page_builder' => 'page_builder',
        'sets' => 'components',
        'typography' => 'typography',
        'collections' => 'my-custom/location',
        // ....
    ]
]
```
Then use the config values in the commands:
```php
$this->viewpath = config('peak-cli.view_folders-page_builder');
// ...
File::put(base_path("resources/views/{$this->viewpath}/_{$this->filename}.antlers.html"), $contents);
```

✅ Configure once and CLI commands works acording to needs
❌ Peak works without any config until now. Not soo nice to have a config just for helper CLI cmds.

I can provide a pull request for both ways, but maybe you can give your opinion on this first?

**Comment by robdekort** [comment]
Thanks for your thorough post and questions. This would also have consequences for other commands like installing the presets. I think I prefer option 2 and not have the config file published by default. So power users can publish a config file and change the settings. 

How about that? If you were to do a PR to add this to all commands that have hard coded folders please target the `feature/v4-support` branch as there are some changes made already there for Statamic v4.

**Comment by exophunk** [comment]
I like the second option too. I've come quite far with the PR, but updating it for all presets as well starts to make it quite complicated, as now all `partial:...` tags need to be dynamic aswell. Lets see

**Reply by robdekort** [reply]
Oh yikes that’s not great indeed.
